File: Presentation Layer (frontend)/Reservation_angular/src/app/service/payment.service.ts
    import { CartItem } from "src/data/cartItem";
    import { HttpClient, HttpHeaders } from '@angular/common/http';
    import { Injectable } from '@angular/core';
    import { Router } from '@angular/router';
    import { Login } from 'src/data/login';
    import { environment } from 'src/environments/environment';
    import { SharedModule } from '../shared/shared.module';
    import jwt_decode from 'jwt-decode';
    import { Register } from 'src/data/register';
    import { ContactUs } from "src/data/contactUs";
    import { ContactUsInfo } from "src/data/contactUsInfo";
    import { Header } from "src/data/header";
import { MenuItemDiscount } from "src/data/menuItemDiscount";
import { Payment } from "src/data/payment";
    @Injectable({
      providedIn: 'root'
    })
    export class PaymentService {
    
      headerDict={
        'Content-Type':'application/json',
        'Accept':'application/json'
        }
     requestOptions={
          headers:new HttpHeaders(this.headerDict)
          }
     constructor(public http:HttpClient,private route:Router) {

      }
     InsertPayment(payment:Payment){
      return  this.http.post<boolean>(environment.apiUrl+"Payment/InsertPayment",payment,this.requestOptions);



     }
     UpdatePayment(payment:Payment){
     return   this.http.put<boolean>(environment.apiUrl+"Payment/UpdatePayment",payment,this.requestOptions);


     }
     DeletePaymentByPaymentId(PaymentId:number){
     return   this.http.delete<boolean>(environment.apiUrl+"Payment/DeletePaymentByPaymentId?PaymentId="+PaymentId,this.requestOptions);


     }
     GetAllPayment(){
     return   this.http.get<any[]>(environment.apiUrl+"Payment/GetAllPayment",this.requestOptions);


     }
     GetPaymentByPaymentId(paymentId:number ){
      return   this.http.get<any[]>(environment.apiUrl+"Payment/GetPaymentByPaymentId?paymentId="+paymentId,this.requestOptions);
 
         
      }   
 
      CountingPaymentGroupByDate(from_Date :Date, to_Date :Date)
      {
        return this.http.get<any[]>(environment.apiUrl+"Payment/CountingPaymentGroupByDate?from_Date="+from_Date+"&to_Date="+to_Date);
      }
    
    CountingPaymentByCustomerIdGroupByDate(from_Date :Date, to_Date :Date, id :number)
  
      {
  
        return this.http.get<any[]>(environment.apiUrl+"Payment/CountingPaymentByCustomerIdGroupByDate?from_Date="+from_Date.toISOString()+"&to_Date="+to_Date.toISOString()+"&id="+id);
      }
    
  
     SumPaymentGroupByDate(from_Date :Date, to_Date :Date)
      {
      
        return this.http.get<any[]>(environment.apiUrl+"Payment/SumPaymentGroupByDate?from_Date="+from_Date.toISOString()+"&to_Date="+to_Date.toISOString());
       }
      SumPaymentByCustomerIdGroupByDate(from_Date :Date, to_Date :Date, id :number)
      {
        return this.http.get<any[]>(environment.apiUrl+"Payment/SumPaymentByCustomerIdGroupByDate?from_Date="+from_Date.toISOString()+"&to_Date="+to_Date.toISOString()+"&id="+id);
      }
     StatisticPaymentByDate(from_Date :Date, to_Date :Date)
      {
        return this.http.get<any[]>(environment.apiUrl+"Payment/StatisticPaymentByDate?from_Date="+from_Date.toISOString()+"&to_Date="+to_Date.toISOString());
      }
    StatisticPaymentByDateAndByCustomerId(from_Date :Date, to_Date :Date, id :number)
      {
        return this.http.get<any[]>(environment.apiUrl+"Payment/StatisticPaymentByDateAndByCustomerId?from_Date="+from_Date.toISOString()+"&to_Date="+to_Date.toISOString()+"&id="+id);
      }
    StatisticPaymentByDateAndGroupByPaymentMethodId(from_Date :Date, to_Date :Date)
      {
        return this.http.get<any[]>(environment.apiUrl+"Payment/StatisticPaymentByDateAndGroupByPaymentMethodId?from_Date="+from_Date.toISOString()+"&to_Date="+to_Date.toISOString());
      }
  
  
    
    }
    

File: Business Logic Layer(Backend)/Reservation/Reservation.Infra/Service/PaymentService.cs
using Reservation.Core.Data;
using Reservation.Core.DTO;
using Reservation.Core.Repository;
using Reservation.Core.Service;
using System;
using System.Collections.Generic;
using System.Text;

namespace Reservation.Infra.Service
{
    public class PaymentService :IPaymentService
  {
        private readonly IPaymentRepositroy _paymentRepositroy;
        public PaymentService(IPaymentRepositroy paymentRepositroy)
        {
            _paymentRepositroy = paymentRepositroy;
        }
        public bool DeletePaymentByPaymentId(decimal PaymentId)
        {
            return _paymentRepositroy.DeletePaymentByPaymentId(PaymentId);
        }

        public List<Payment> GetAllPayment()
        {
            return _paymentRepositroy.GetAllPayment();
        }

        public bool InsertPayment(Payment payment)
        {
            return _paymentRepositroy.InsertPayment(payment);
        }
        public List<Payment> GetPaymentByPaymentId(decimal paymentId)
        {
            return _paymentRepositroy.GetPaymentByPaymentId(paymentId);
        }
        public bool UpdatePayment(Payment payment)
        {
            return _paymentRepositroy.UpdatePayment(payment);
        }

    public List<PaymentDto> CountingPaymentGroupByDate(DateTime from_Date, DateTime to_Date)
    {
      return _paymentRepositroy.CountingPaymentGroupByDate(from_Date, to_Date);
    }

    public List<PaymentDto> CountingPaymentByCustomerIdGroupByDate(DateTime from_Date, DateTime to_Date, decimal id)

    {

      return _paymentRepositroy.CountingPaymentByCustomerIdGroupByDate(from_Date,to_Date,id);
    }
    

    public List<PaymentDto> SumPaymentGroupByDate(DateTime from_Date, DateTime to_Date)
    {
      return _paymentRepositroy.SumPaymentGroupByDate(from_Date, to_Date);
    }

    public List<PaymentDto> SumPaymentByCustomerIdGroupByDate(DateTime from_Date, DateTime to_Date, decimal id)
    {
      return _paymentRepositroy.SumPaymentByCustomerIdGroupByDate(from_Date, to_Date,id);
    }

    public List<PaymentDto> StatisticPaymentByDate(DateTime from_Date, DateTime to_Date)
    {
      return _paymentRepositroy.StatisticPaymentByDate(from_Date, to_Date);
    }

    public List<PaymentDto> StatisticPaymentByDateAndByCustomerId(DateTime from_Date, DateTime to_Date, decimal id)
    {
      return _paymentRepositroy.StatisticPaymentByDateAndByCustomerId(from_Date, to_Date, id);
    }

    public List<PaymentDto> StatisticPaymentByDateAndGroupByPaymentMethodId(DateTime from_Date, DateTime to_Date)
    {
      return _paymentRepositroy.StatisticPaymentByDateAndGroupByPaymentMethodId(from_Date,to_Date);
    }
  }
}
